easybuild/easyblocks/c/cp2k.py

```diff
@@ -668,9 +668,15 @@ def build_step(self):
         run_cmd(cmd + " clean", log_all=True, simple=True, log_output=True)
 
         # build and install
+        # compile regularly first with the default make target
+        # and only then build the library
+        run_cmd(cmd, log_all=True, simple=True, log_output=True)
+
+        # build as a library
         if self.cfg['library']:
             cmd += ' libcp2k'
-        run_cmd(cmd + " all", log_all=True, simple=True, log_output=True)
+            run_cmd(cmd, log_all=True, simple=True, log_output=True)
```

Comment:
This should probably just be:

```python
if self.cfg['library']:
    run_cmd(cmd + " libcp2k", log_all=True, simple=True, log_output=True)
```

So, don't change the `cmd` local variable.
It doesn't matter much right now, since we're at the end of the `build_step` method here, but that may change in the future, and then we may be surprised by getting a changed `cmd` value after this...

Reply:
Makes sense, I thought about changing it as well.